src/database/queries.rs

```diff
@@ -25,76 +25,112 @@ use sp_runtime::traits::Block as BlockT;
 use sqlx::PgConnection;
 use std::convert::TryFrom;
 
-/// get missing blocks from relational database as a stream
+/// Return type of queries that `SELECT version`
+struct Version {
+    version: i32,
+}
+
+/// Return type of querys that `SELECT missing_num ... FROM ... GENERATE_SERIES(a, z)`
+pub struct Series {
+    missing_num: Option<i32>,
+}
+
+/// Return type of querys that `SELECT MAX(int)`
+struct Max {
+    max: Option<i32>,
+}
+
+/// Return type querys that `SELECT EXISTS`
+struct DoesExist {
+    exists: Option<bool>,
+}
+
+// Return type of querys that `SELECT block_num`
+struct BlockNum {
+    block_num: i32,
+}
+
+// Return type of querys that `SELECT data`
+struct Bytes {
+    data: Vec<u8>,
+}
+
+///  Get the `block_num` of missing blocks from relational database as a stream
 #[allow(unused)]
 pub(crate) fn missing_blocks_stream(
     conn: &mut PgConnection,
-) -> impl Stream<Item = Result<(i32,)>> + Send + '_ {
-    sqlx::query_as::<_, (i32,)>(
-        "SELECT generate_series
-        FROM (SELECT 0 as a, max(block_num) as z FROM blocks) x, generate_series(a, z)
+) -> impl Stream<Item = Result<Series>> + Send + '_ {
+    sqlx::query_as!(
+        Series,
+        "SELECT missing_num
+        FROM (SELECT 0 as zero, MAX(block_num) as max FROM blocks) zero_to_max, 
+            GENERATE_SERIES(zero, max) as missing_num
         WHERE
-        NOT EXISTS(SELECT id FROM blocks WHERE block_num = generate_series)
-        ORDER BY generate_series ASC
+        NOT EXISTS(SELECT id FROM blocks WHERE block_num = missing_num)
+        ORDER BY missing_num ASC
         ",
     )
     .fetch(conn)
     .map_err(Error::from)
 }
 
-/// get missing blocks from relational database
+/// Get the `block_num` of missing blocks from relational database
 #[allow(unused)]
```

Comment:
We can just remove these methods, they're holdovers that I kept for reference in case I wanted to use a stream with one of the queries while testing using a stream vs returning as a vec, but are not used for anything currently.

Reply:
Removed